source/includes/data-modeling/field-types.rb

```diff
@@ -0,0 +1,247 @@
+# start-define-fields
+class Person
+  include Mongoid::Document
+  field :name, type: String
+  field :date_of_birth, type: Date
+  field :weight, type: Float
+end
+# end-define-fields
+
+# start-define-untyped
+class Product
+  include Mongoid::Document
+
+  field :name, type: String
+  field :properties
+end
+# end-define-untyped
+
+# start-untyped
+product = Product.new(properties: "color=white,size=large")
+# properties field saved as String: "color=white,size=large"
+
+product = Product.new(properties: {color: "white", size: "large"})
+# properties field saved as Object: {:color=>"white", :size=>"large"}
+# end-untyped
+
+# start-stringified-symbol
+class Post
+  include Mongoid::Document
+  
+  field :status, type: StringifiedSymbol
+end
+  
+# Save status as a symbol
+post = Post.new(status: :hello)
+# status is stored as "hello" on the database, but returned as a Symbol
+post.status
+# Outputs: :hello
+
+# Save status as a string
+post = Post.new(status: "hello")
+# status is stored as "hello" in the database, but returned as a Symbol
+post.status
+# Outputs: :hello
+# end-stringified-symbol
+
+# start-hash
+class Person
+  include Mongoid::Document
+  field :first_name
+  field :url, type: Hash
+end
+
+person = Person.new(url: {'home_page' => 'http://www.homepage.com'})
+# end-hash
+
+# start-time
+class Voter
+  include Mongoid::Document
+  
+  field :registered_at, type: Time
+end
+  
+Voter.new(registered_at: Date.today)
+# end-time
+
+# start-datetime
+class Ticket
+  include Mongoid::Document
+  field :purchased_at, type: DateTime
+end
+# end-datetime
+
+# start-datetime-int
+ticket.purchased_at = 1544803974
+ticket.purchased_at
+# Outputs: Fri, 14 Dec 2018 16:12:54 +0000
+# end-datetime-int
+
+# start-datetime-string
+ticket.purchased_at = 'Mar 4, 2018 10:00:00 +01:00'
+ticket.purchased_at
+# Outputs: Sun, 04 Mar 2018 09:00:00 +0000
+# end-datetime-string
+
+# start-timestamps
+class Post
+  include Mongoid::Document
+  include Mongoid::Timestamps
+end
+# end-timestamps
+
+# start-timestamps-specific
+class Post
+  include Mongoid::Document
+  include Mongoid::Timestamps::Created
+end
+
+class Post
+  include Mongoid::Document
+  include Mongoid::Timestamps::Updated
+end
+# end-timestamps-specific
+
+# start-timestamps-disable
+post.timeless.save
+# end-timestamps-disable
+
+# start-timestamps-short
+class Band
```

Comment:
S: consider using the Post class for consistency with the prev examples in this section